contracts/vending-minter/src/contract.rs

```diff
@@ -590,6 +592,45 @@ fn random_mintable_token_mapping(
     Ok(TokenPositionMapping { position, token_id })
 }
 
+pub fn execute_update_mint_price(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    price: u128,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    let mut config = CONFIG.load(deps.storage)?;
+    if info.sender != config.extension.admin {
+        return Err(ContractError::Unauthorized(
+            "Sender is not an admin".to_owned(),
+        ));
+    }
+    // TODO add test
+    // If current time is after the stored start time, only allow lowering price
+    if env.block.time >= config.extension.start_time && price >= config.mint_price.amount.u128() {
+        return Err(ContractError::UpdatedMintPriceTooHigh {
+            allowed: config.mint_price.amount.u128(),
+            updated: price,
+        });
+    }
+
+    // TODO add test
+    // If price is greater than initial price, return error
+    if price > config.extension.initial_price.amount.u128() {
+        return Err(ContractError::UpdatedMintPriceTooHigh {
+            allowed: config.extension.initial_price.amount.u128(),
+            updated: price,
+        });
+    }
```

Comment:
This won't let update price above the initial price even if the public start time hasn't started.

Reply:
is there a point to `initial_price` if you can set it above or below before the sale and it's not used after sale?